public/static/docs/user-guide/setup-google-drive-remote.md

```diff
@@ -0,0 +1,114 @@
+
+⚠️ It's safe to share **client ID** and **client secret** among your team. These
+credentials are used to
+[authorize](https://developers.google.com/drive/api/v2/about-auth) DVC to use
+your Google Drive.
+
+> Note that Google Drive API has usage limits/quotas per credentials in a
+> _project_ (which you can review in the
+> [OAuth consent screen](https://console.developers.google.com/apis/credentials/consent)).
+> Please keep this in mind when sharing them, or you may
+> [exceed the limits](https://developers.google.com/drive/api/v2/handle-errors?hl=ro#resolve_a_403_error_usage_limit_exceeded).
+
+## URL format for Google Drive remotes
+
+A GDrive DVC remote URLs is needed for the `dvc remote add` command. It can be
+constructed with a _base_, and an **optional** _path_ to a folder i.e.
+`gdrive://<base>/path/to/folder`. The base can be one of:
+
+1. `root` - indicates your topmost Google Drive directory.
+
+  ```dvc
+  $ dvc remote add -d mygdroot gdrive://root
+  $ dvc remote add mygdfolder gdrive://root/path/to/folder
+  ```
+
+2. Google Drive _Folder ID_
+
+  To obtain the unique _Folder ID_ in question, navigate into that folder in
+  your web browser, and find it in the address bar. For example, for
+  `https://drive.google.com/drive/folders/0AIac4JZqHhKmUk9PDA`:
+
+  ```dvc
+  $ dvc remote add mygdfolder gdrive://0AIac4JZqHhKmUk9PDA
+  $ dvc remote add mygdsubfolder \
+                   gdrive://0AIac4JZqHhKmUk9PDA/sub/path
+  ```
+
+  > Note that
+  > [shared drives](https://developers.google.com/drive/api/v2/about-shareddrives)
+  > can be referenced by _Folder ID_ in the same way.
+
+3. (rare) `appDataFolder` - special
+   [application-specific data](https://developers.google.com/drive/api/v2/appdata)
+   folder only accessible by your application and hidden from the user.
+
+  ```dvc
+  $ dvc remote add mygdappata gdrive://appDataFolder
+  ```
+
+## Configure the GDrive remote
+
+Use the `dvc remote modify` command to set the credentials for each `gdrive://`
+remote, for example:
+
+```dvc
+$ dvc remote add -d mygdfolder gdrive://root/path/to/folder
+$ dvc remote modify mygdfolder gdrive_client_id <client ID>
+$ dvc remote modify mygdfolder gdrive_client_secret <client secret>
+```
+
+## Authorization
+
+On the first usage of a GDrive [DVC remote](/doc/command-reference/remote),
+you'll be prompted to visit a web page to generate an access token, which will
+require you to sign into the corresponding Google account. The sign-in process
+will guide you through granting the necessary access permissions. On success,
+the token data will be cached in a Git-ignored directory located in
+`.dvc/tmp/gdrive-user-credentials.json`.
+
+⚠️ In order to prevent unauthenticated access to your Google Drive, do not share
+access token data with others. Each team member should have
+[separate access](https://support.google.com/drive/answer/7166529) to Google
+Drive, and generate their own token.
```

Comment:
> Each team member should have [separate access](https://support.google.com/drive/answer/7166529) to Google Drive, and generate their own token.

This sounds technically a little bit strange to me. Just can't understand it :) I might suggest just to remove this since the main idea of this warning is to prevent sharing of generated `.dvc/tmp/gdrive-user-credentials.json` file.

Reply:
Hmmm. I guess we could just shorten it to "Each team member should generate their
own tokens."

But the point is to say that each user needs access to a folder so they can connect their DVC projects to it. And https://support.google.com/drive/answer/7166529 is an article by Google on sharing Google Drive folder access.  Am I confused?

Reply:
https://support.google.com/drive/answer/7166529 article doesn't look relevant (it describes folders shared through GDrive's web ui) to what happens there ( Google generates access token data which can be used to access all user's files on user's behalf ). Short version sounds better, thanks

Reply:
OK maybe I need to rewrite or remove this but my point is that each user generates their own token right? Even if they all use the same client ID/secret for OAuth. But this means each token has different access. If it's someone else's folder, they have to give your user permissions first (via the web UI being the simplest way).

Reply:
these tokens in our case are equivalent to giving login/password to your GDrive - so they are personal. They are in the tmp dir for that reason (Gitignored). They should not be shared. 

It also means that they give DVC access to the same directories/shared drives/etc as you would get with your login/password. Again, think about them as SSH keys (auto password/login)

`Each team member should generate their own tokens` - in 99% of cases sounds correct.

Reply:
Going to replace last sentence with `Each team member should generate their own tokens`.